librosa/decompose.py

```diff
@@ -322,7 +357,7 @@ def hpss(S, kernel_size=31, power=2.0, mask=False):
         perc[zero_harm & zero_perc] = 0.5
 
         # Compute harmonic mask
-        mask_harm = harm / (harm + perc)
-        mask_perc = perc / (harm + perc)
+        mask_harm = (harm >  perc * margin_harm).astype(float)
```

Comment:
This looks like it can only produce binary mask matrices, rather than wiener filtering soft-masks.

I think what you need to do here is rework the mask logic so that a third soft-mask (`res`) captures all energy not assigned to `harm` or `perc`.  Then you would get `mask_harm = harm / (harm + perc + res)`